src/CosmosCache.cs

```diff
@@ -115,8 +115,11 @@ public async Task<byte[]> GetAsync(string key, CancellationToken token = default
                 }
                 catch (CosmosException cosmosException) when (cosmosException.StatusCode == HttpStatusCode.PreconditionFailed)
                 {
-                    // Race condition on replace, we need to get the latest version of the item
-                    return await this.GetAsync(key, token).ConfigureAwait(false);
+                    if (this.options.RetrySlidingExpirationUpdates)
```

Comment:
It should be possible to cover this with unit tests (in the UT project) by mocking the client to return a 412 on the write operation and making sure the mock is invoked once